amplipi/ctrl.py

```diff
@@ -585,17 +610,22 @@ def set_zone(self, zid, update: models.ZoneUpdate, force_update: bool = False, i
         zone.name = name
         zone.disabled = disabled
 
-        # any disabled zone should not be able to play anything, mute it to be sure
-        if zone.disabled and not mute:
+        # parse and check the source id
+        sid = utils.parse_int(source_id, range(models.SOURCE_DISCONNECTED, models.MAX_SOURCES))
+
+        # any zone disabled by source disconnection or a 'disabled' flag should not be able to play anything
+        implicit_mute = zone.disabled or sid == models.SOURCE_DISCONNECTED
```

Comment:
It would be fairly trivial to support zones being physically disconnected from all sources in firmware, is that something we should add to the todo list or is this good enough?

Reply:
Definitely add it to the todo list.